### ISSUE TITLE
Please add doc on how to provide data on command line

### ISSUE THREAD

**Issue by mjmare** [Fizzadar/pyinfra#566]
This section in the docs https://docs.pyinfra.com/en/1.x/deploys.html#data-hierarchy suggests that it is possible to provide host data on the CLI. 
However I have not found a way to do this. This is my best guess:
```
pyinfra inventories/post_boot.py post_boot.py NEW_READER_NO=1
```
I'd expect to use this in a deploy as host.data.NEW_READER_NO.

What I get is a weird error:
```
--> Loading config...
--> An unexpected internal exception occurred:

  File "/Users/mjm/Dropbox/Devel/ACHN/inuit_cardreader/venv/lib/python3.9/site-packages/pyinfra/pseudo_modules.py", line 35, in __getattr__
    return getattr(self._base_module, key)
Traceback (most recent call last):
  File "/Users/mjm/Dropbox/Devel/ACHN/inuit_cardreader/venv/lib/python3.9/site-packages/pyinfra_cli/util.py", line 131, in get_operation_and_args
    op_module = import_module('pyinfra.operations.{0}'.format(op_module))
  File "/Users/mjm/.asdf/installs/python/3.9.2/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'pyinfra.operations.post_boot'

During handling of the above exception, another exception occurred:

AttributeError: type object 'Host' has no attribute 'data'

--> The full traceback has been written to pyinfra-debug.log
--> If this is unexpected please consider submitting a bug report on GitHub, for more information run `pyinfra --support`.
```

In case you were wondering why would anyone want to do this? I'd like to prompt the user for some input. This is not possible with PyInfra (and I can understand why). So I have a script that ask for the data, which I then want to provide to the pyInfra deploy.

**Comment by Fizzadar** [comment]
Hi @mjmare! Unfortunately this isn't (currently) possible - the "override" data specifically applies to things like `--port`, etc. I will update the doc to make this explicit because it does indeed suggest anything is possible!

I'd also like to make it possible to pass data via a CLI flag! Potentially in the form `--data key=value ...'.

**Comment by mjmare** [comment]
Thank you!

**Comment by Fizzadar** [comment]
Implemented + released in v1.5.

**Comment by mjmare** [comment]
Thank you. This will make scripting much easier.
